clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -145,8 +148,32 @@ public void start() throws Exception {
         if (connectionState != HubConnectionState.DISCONNECTED) {
             return;
         }
+        if (!skipNegotiate) {
+            int negotiateAttempts = 0;
+            do {
+                if(negotiateAttempts > 0 && this.negotiateResponse.accessToken != null){
+                    this.accessToken = negotiateResponse.accessToken;
+                    this.negotiateResponse = Negotiate.processNegotiate(url, this.negotiateResponse.accessToken);
+                    this.url = this.url + "&id=" + negotiateResponse.connectionId + "&access_token=" + this.accessToken;
```

Comment:
Should I just store the `accessToken`  in the `Authorization` header instead?

Reply:
Yes

Reply:
This will just keep appending `&id` and `&access_token` as the client keeps getting redirected. I'm not sure the logic needs to be different based on which attempt we're on. The loop [in the C# client
](https://github.com/aspnet/SignalR/blob/release/2.2/src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs#L314) doesn't really do that. The initial URL and initial Access Token aren't relevant at all once the negotiate response comes in (https://github.com/aspnet/SignalR/issues/2859 notwithstanding, but we should come back to that)